### PR TITLE
Added focus indicator to RadioButton

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Added focus indicator to the radio button component.
I also changed the RadioButton/Children story to show how focus indicator could be implemented with custom children.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
Closes https://github.com/grommet/grommet/issues/6328

#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

**Comment by britt6612** [comment]
Screenshot 2024-11-20 at 6 28 18 PM


we need to add the check if RadioButton is within a formfield 

we have similar things in checkbox and RadioButtonGroup

```
      // when contained in a FormField, focusIndicator = false,
      // so that the FormField has focus style. However, we still
      // need to visually indicate when a CheckBox is active.
      // If focus = true but focusIndicator = false,
      // we will apply the hover treament.

```

**Review by britt6612 — COMMENTED** [review]
behavior looks correct though when outside `formfield`

**Comment by jcfilben** [comment]
behavior looks correct though when outside formfield

Oops good catch, realized I only tested RadioButtonGroup in a Form and didn't test just a RadioButton

File: src/js/components/RadioButton/RadioButton.js [hunk 1]
```diff
@@ -71,6 +73,12 @@ const RadioButton = forwardRef(
         }}
         focus={focus}
         focusIndicator={focusIndicator}
+        onFocus={() => {
+          setFocus(true);
+        }}
+        onBlur={() => {
+          setFocus(false);
+        }}
```
Comment (MikeKingdom) [inline src/js/components/RadioButton/RadioButton.js:81]: To match the handlers below
```suggestion
        onFocus={() => setFocus(true)}
        onBlur={() => setFocus(false)}
```

**Comment by britt6612** [comment]
> > behavior looks correct though when outside formfield
> 
> Oops good catch, realized I only tested RadioButtonGroup in a Form and didn't test just a RadioButton

all good :)

**Review by MikeKingdom — COMMENTED** [review]
Looking pretty good.

Should we tweak the RadioButtonGroup/Children story so you get a real focus indicator also?

**Review by MikeKingdom — APPROVED** [review]
Looks good!

**Review by britt6612 — APPROVED** [review]
LGTM

**Pull request merged**